Fixed error handling of nonexisting columns

pl0xz0rz opened the pull request:
This PR should fix #205 by raising a KeyError when an invalid column is used from a histogram.

However, a bug was discovered in joins when making the changes - columns in joins with implicit sources aren't added to the dependency tree - the fix for that bug is more complex, and the bug has a workaround by chaining table names with a dot, so it will probably be implemented in a later PR.

Possibly also fixes #121 by fixing inconsistent usage of "edges" between 1D and ND

miranov25 commented:
# Running full test parallel suit - bugs found

* new check cause failure of the FAILED test_bokehClientHistogram.py test

New checks found a bug in the previous test which was not seen in Python, only in javascript.
2 problems were identified:

- problem in test - wrong variables
- inconsistent usage of the "edges" between 1D and ND
  - as the user did not use them can be fixed by non-compatible change 



```
 makePyTest 6 test6 /lustre/alice/users/miranov/tests/RootInteractive alice
```

### error summary

```
-- Docs: https://docs.pytest.org/en/stable/warnings.html
---------------------------------------------------------------------------------------------- JSON report ----------------------------------------------------------------------------------------------
report saved to: test6.json
======================================================================================== short test summary info ========================================================================================
FAILED test_bokehDrawSA.py::testBokehDrawArraySA_tree - NameError: name 'figureLayout' is not defined
FAILED test_bokehClientHistogram.py::testBokehClientHistogram - KeyError: 'Column A not found in histogram histoA'
========================================================================== 2 failed, 35 passed, 9 warnings in 84.08s (0:01:24) ==========================================================================
```

miranov25 commented:
After  the fix there are more problems in other tests:

```
source $RootInteractive/tests/pytest.sh 
makePyTest 6 test6 /lustre/alice/users/miranov/tests/RootInteractive alice 
```

miranov25 commented:
## Still problem:
```
======================================================================================== short test summary info ========================================================================================
FAILED test_bokehDrawSA.py::testBokehDrawArraySA_tree - NameError: name 'figureLayout' is not defined
FAILED test_bokehClientHistogram.py::testBokehClientHistogram - KeyError: 'Column A not found in histogram histoA'
========================================================================== 2 failed, 35 passed, 9 warnings in 72.48s (0:01:12) ==========================================================================
```

miranov25 commented:
### The check works only for the "new interface":
- "new interface" (since 6 months) data source is explicit
- "old interface"  (still working) - data source is derived by string parsing
-- not possible to check the consistency before parsing

pl0xz0rz commented:
Fixing inconsistent usage of "edges" between 1D and ND also fixed test_bokehClientHistogram.py::testBokehClientHistogram crashing on the client with bars visualization and old interface, and also fixed 1D histogram tooltips.

Looking at the output from test_bokehClientHistogramWeight.py output, the data source names on figure labels seemed to be shifted back by one figure - this has been fixed, to be tested against previous version to see if this was the case before

FAILED test_bokehClientHistogram.py::testBokehClientHistogram - KeyError: 'Column A not found in histogram histoA'

This test should pass now, by adding a special null data source that has no columns, and is ignored for the dependency tree.

FAILED test_bokehDrawSA.py::testBokehDrawArraySA_tree - NameError: name 'figureLayout' is not defined

This test seems to be already failing on master

miranov25 commented:
Thanks . There remained only one failure, which was here already before:
FAILED test_bokehDrawSA.py::testBokehDrawArraySA_tree

Lets fic it in the next commit

miranov25 commented:
Merging. Lets fix remaining test in separate pull request
